packages/base/skill-card.gts

```diff
@@ -1,11 +1,55 @@
+import BooleanField from './boolean';
+import CodeRefField from './code-ref';
 import MarkdownField from './markdown';
-import { CardDef, Component, field, contains } from './card-api';
+import StringField from './string';
+import {
+  CardDef,
+  Component,
+  field,
+  FieldDef,
+  contains,
+  containsMany,
+} from './card-api';
 import RobotIcon from '@cardstack/boxel-icons/robot';
 
+function djb2_xor(str: string) {
+  let len = str.length;
+  let h = 5381;
+
+  for (let i = 0; i < len; i++) {
+    h = (h * 33) ^ str.charCodeAt(i);
+  }
+  return (h >>> 0).toString(16);
+}
+
+export class CommandField extends FieldDef {
+  static displayName = 'CommandField';
+  @field codeRef = contains(CodeRefField, {
+    description: 'An absolute code reference to the command to be executed',
+  });
+  @field requiresApproval = contains(BooleanField, {
+    description:
+      'If true, this command will require human approval before it is executed in the host.',
+  });
+
+  @field functionName = contains(StringField, {
+    description: 'The name of the function to be executed',
+    computeVia: function (this: CommandField) {
+      if (!this.codeRef?.module || !this.codeRef?.name) {
+        return '';
+      }
+
+      const hashed = djb2_xor(`${this.codeRef.module}#${this.codeRef.name}`);
+      return `${this.codeRef.name}_${hashed.slice(0, 4)}`;
```

Comment:
Relying on the name without the module seems problematic to me. We don't really want to try to enforce how userland commands are organized do we?